### PR TITLE
feat: root README

### LINKED ISSUE #6: Root README
While the other two READMEs serve as introductions to their respective frameworks and their important files, this README is meant to be an introduction to the starter pack as a whole. It should discuss at a high level what full stack web applications are and introduce what frameworks are being used at a high level.

### PR DESCRIPTION
**Pull request by samderanova**
This PR contains a draft of what should go in the root README.md.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,56 @@
-# React + FastAPI Full Stack Starter Pack
+# Starter Pack Series: Full Stack Web Application
```
Comment (taesungh) [inline README.md]: Chore [repeating]: full-stack as an adjective should be hyphenated.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,56 @@
...
 
+## Introduction
+
+Welcome to the starter pack series on full stack web applications! This
```
Comment (taesungh) [inline README.md]: Chore: starter-pack as an adjective (but not as a noun) should be hyphenated.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,56 @@
...
+repository serves as an introduction to how full stack web applications work
+and how to leverage frameworks to make it easier to build one.
+
+## What are Frontends and Backends?
+
+A frontend is what appears on a webpage, while a backend is what happens
+behind the scenes, like fetching the data that will eventually be shown on the
```
Comment (taesungh) [inline README.md]: Thought: "providing" might be more appropriate than "fetching".

File: README.md [hunk 4]
```diff
@@ -1,2 +1,56 @@
...
+webpage.
+
+Frontends generally consist of HTML, CSS, and JavaScript. While you can code
+directly using these languages, there are other frameworks such as React that
+allow you to do the same faster and more efficiently.
+
+Backends are much more flexible in that all you need is an HTTP server that the
+client can send
+[different types of requests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers)
+to, along with your own custom routes (i.e. http://localhost:5000/my-route) each
```
Comment (taesungh) [inline README.md]: Nitpick: "e.g." would be more appropriate than "i.e."

File: README.md [hunk 5]
```diff
@@ -1,2 +1,56 @@
...
+of which can perform its own action, such as inserting to a database vs. deleting
+a record from a database. That being said, there are many libraries, such as
+FastAPI (Python) and Express (JavaScript) that have different ways of doing this.
+
+## What is a Full Stack Web App?
+
+A full stack web application generally consists of a frontend client and a
+backend server. For example, let's say you're trying to access a profile page
+on some web application and the data is stored on a database somewhere. When
+the page is loading, the client is sending a request to the backend so the
```
Comment (taesungh) [inline README.md]: Suggestion: present tense may sound smoother than progressive
> When the page loads, the client sends ...

File: README.md [hunk 6]
```diff
@@ -1,2 +1,56 @@
...
+backend, in turn, can retrieve the data from the database. The server will then
```
Comment (taesungh) [inline README.md]: Repeated use of backend seems disruptive. Consider
> to the backend which in turn retrieves ...

File: README.md [hunk 7]
```diff
@@ -1,2 +1,56 @@
...
+A frontend is what appears on a webpage, while a backend is what happens
+behind the scenes, like fetching the data that will eventually be shown on the
+webpage.
+
+Frontends generally consist of HTML, CSS, and JavaScript. While you can code
+directly using these languages, there are other frameworks such as React that
+allow you to do the same faster and more efficiently.
+
+Backends are much more flexible in that all you need is an HTTP server that the
+client can send
```
Comment (taesungh) [inline README.md:20]: Aside: interesting to think about how the lines between frontend and backend are melded when considering older templated models such as PHP and modern server-side models such as Next.js, but in the statically-deployed SPA model the API needs to be separate.

File: README.md [hunk 8]
```diff
@@ -1,2 +1,56 @@
...
+backend server. For example, let's say you're trying to access a profile page
+on some web application and the data is stored on a database somewhere. When
+the page is loading, the client is sending a request to the backend so the
+backend, in turn, can retrieve the data from the database. The server will then
+send the data back to the frontend so that the frontend can display it to the
+user. This interaction is just one example of how a frontend and backend can work
+together to provide a great and scalable experience for its users.
+
+**Important:** You might be wondering why you can't just query the database
+directly from the frontend. Databases require sensitive credentials for access and
```
Comment (taesungh) [inline README.md:40]: Praise: good to include this elaboration.

**Review by taesungh — COMMENTED** [review]
Thanks for including, some minor notes/corrections.

File: README.md (hunk 7 above)
Comment (samderanova) [reply README.md:20]: Is this something we should include in the README then or not since it's not too relevant to the SPA structure of this starter pack?

Comment (taesungh) [reply README.md:20]: Just an observation on how a lot of our thought process and explanation is based on the separation concept.

**Review by taesungh — APPROVED** [review]

**Pull request merged**